etc/submit.sh

```diff
@@ -10,37 +10,100 @@ sed -e 's/^/export /' startup_environment.sh > tmp_env.sh
 mv tmp_env.sh startup_environment.sh
 export JOBSTARTDIR=$PWD
 
+# Saving START_TIME and when job finishes END_TIME.
+# Multiple checks are done:
+# 1) Check submit.sh bootstrap exit code.
+#    If it is not 0 sleep MIN_JOB_RUNTIME_SECONDS and in case it is 0, force to sleep 5 mins total runtime
+#    If this bootstrap script is exiting non 0 exit code, something is wrong on this WN...
+# 2) If Job exited 0, do not sleep any timeout
+# 3) Otherwise, sleep up to MIN_JOB_RUNTIME_TIME_SECONDS seconds of runtime.
+#    By default MIN_JOB_RUNTIME_TIME_SECONDS will be set to 0, if this is not passed through agent schedd configuration
+# Fast finishing jobs (failed specially) might overload schedulers and 
+# restarting jobs on same bad WN leads to more failures
+START_TIME=$(date +%s)
+function finish {
+  exitCode=$?
+  echo "======== WMAgent final job runtime checks STARTING at $(TZ=GMT date) on $(hostname) ========"
+  if [ $exitCode -ne 0 ];
+  then
+      if [ $MIN_JOB_RUNTIME_TIME_SECONDS -eq 0 ];
+      then
+          MIN_JOB_RUNTIME_TIME_SECONDS=300
+      fi
+  elif [ $jobrc -eq 0 ];
+  then
+      MIN_JOB_RUNTIME_TIME_SECONDS=0
+  fi
+  END_TIME=$(date +%s)
+  DIFF_TIME=$((END_TIME-START_TIME))
+  echo "$(TZ=GMT date): Job Runtime in seconds: " $DIFF_TIME
+  echo "$(TZ=GMT date): Job bootstrap script exited: " $exitCode
+  echo "$(TZ=GMT date): Job execution exited: " $jobrc
+  if [ $DIFF_TIME -lt $MIN_JOB_RUNTIME_TIME_SECONDS ];
+  then
+    SLEEP_TIME=$((MIN_JOB_RUNTIME_TIME_SECONDS - DIFF_TIME))
+    echo "$(TZ=GMT date): Job runtime is less than $MIN_JOB_RUNTIME_TIME_SECONDS seconds. Sleeping " $SLEEP_TIME
+    sleep $SLEEP_TIME
+  fi
+  echo "======== WMAgent final job runtime checks FINISH at $(TZ=GMT date) on $(hostname) ========"
+}
+# Trap all exits and execute finish function
+trap finish EXIT
+
 # should be a bit nicer than before
-echo "WMAgent bootstrap : `date -u` : starting..."
+echo "======== WMAgent bootstrap STARTING at $(TZ=GMT date) on $(hostname) ========"
+echo "Local time : $(date)"
+echo "Current system : $(uname -a)"
+echo "Arguments are $@"
 
 # We need to create the expected output file in advance just in case
 # some problem happens during the job bootstrap
 outputFile="Report.0.pkl Report.1.pkl Report.2.pkl Report.3.pkl"
-if [ -n "$_CONDOR_JOB_AD" ]; then
-    outputFile=`grep ^TransferOutput $_CONDOR_JOB_AD | awk -F'=' '{print $2}' | sed 's/\"//g'`
+MIN_JOB_RUNTIME_TIME_SECONDS=0
```

Comment:
Set it to 300 secs please and rename it to `WMA_MIN_JOB_RUNTIMESECS`, just to avoid any var name collision.